cmd/kafkalock/kafkalock.go

```diff
@@ -0,0 +1,143 @@
+package main
+
+import (
+	"bufio"
+	"flag"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/Shopify/sarama"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+
+	"github.com/grongor/go-kafka-lock/kafkalock"
+)
+
+var (
+	bootstrapServers = flag.String("bootstrapServers", "", "List of Kafka bootstrap servers, separated by comma")
+	kafkaVersion     = flag.String("kafkaVersion", "2.2.0", "Version of the Kafka cluster")
+	overrideTimeouts = flag.Bool("overrideTimeouts", true, "Whether or not to use recommended timeouts")
+	topic            = flag.String("topic", "", "Topic used for the lock, eg. lock.someAppName")
+	maxHolders       = flag.Int("maxHolders", 1, "Maximum allowed number of lock holders, usually just 1 (exclusive lock)")
+	topicPolicy      = flag.String("topicPolicy", "validateOnly", "Topic validation/creation policy, possible values: noValidation, validateOnly, create, dropAndCreate")
+	logKafka         = flag.Bool("logKafka", false, "Enable Kafka logging (logs from sarama library)")
+	debug            = flag.Bool("debug", false, "Enable debug logging")
+)
+
+func main() {
+	flag.Parse()
+
+	var logger *kafkalock.Logger
+	if *debug {
+		logger = kafkalock.NewLogger()
+	} else {
+		logger = kafkalock.NewLogger()
+		noDebugLogger := logger.Desugar().WithOptions(zap.WrapCore(func(core zapcore.Core) zapcore.Core {
+			return zapcore.NewCore(
+				zapcore.NewConsoleEncoder(zap.NewDevelopmentEncoderConfig()),
+				zapcore.AddSync(os.Stderr),
+				zapcore.InfoLevel,
+			)
+		}))
+		logger = &kafkalock.Logger{SugaredLogger: noDebugLogger.Sugar()}
+	}
+
+	if *logKafka {
+		logger.EnableKafkaLogging()
+	}
+
+	if *bootstrapServers == "" {
+		logger.Fatal("You must provide at least one bootstrap server")
+	}
+
+	var err error
+
+	kafkaConfig := sarama.NewConfig()
+	kafkaConfig.Net.SASL.Enable = true
+	kafkaConfig.Net.SASL.Mechanism = sarama.SASLTypePlaintext
+	kafkaConfig.Net.SASL.Version = sarama.SASLHandshakeV1
+	kafkaConfig.Net.SASL.User = "admin"
+	kafkaConfig.Net.SASL.Password = "K5yhbK4fq9KBJomt"
```

Comment:
?

Reply:
:D fuck me, forgot some dev code :) invalidated, gonna remove it :)